core/domain/stats_domain.py

```diff
@@ -544,51 +545,55 @@ def to_dict(self):
         }
 
     @classmethod
-    def from_dict(cls, playthrough_dict):
-        """Returns a Playthrough object from a dict.
+    def validate_input_data(cls, playthrough_data):
```

Comment:
validate_playthrough_dict is better, I think. Our convention is to use _dict for dicts; anything without that suffix is considered to be a domain object.

EDIT: actually, let's not pull this out into a separate method, but keep it in from_dict(). Ditto elsewhere. I think it doesn't make sense to have it as a standalone method given that we don't want other functions to call it (and also don't want to encourage passing dicts around the codebase).

Reply:
Makes sense. Forgot that python doesn't have access modifiers.